apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/ClientBuilder.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Note: this will be in GAX, with the right copyright notice.
+
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Base class for API-specific builders.
+    /// </summary>
+    /// <typeparam name="TClient">The type of client created by this builder.</typeparam>
+    public abstract class ClientBuilder<TClient>
+    {
+        /// <summary>
+        /// The endpoint to connect to, or null to use the default endpoint.
+        /// </summary>
+        public ServiceEndpoint Endpoint { get; set; }
+
+        /// <summary>
+        /// The scopes to use, or null to use the default scopes.
+        /// </summary>
+        public IReadOnlyList<string> Scopes { get; set; }
```

Comment:
nit: Makes no sense for this to be readonly now.

Reply:
I think it still does, but for a different reason: we don't *need* to be able to write to it. We could potentially make it `IEnumerable<string>` instead, admittedly.